### PR TITLE
Update Collector configuration telemetry section

### PR DESCRIPTION
**Pull request by brightzheng100**
Update Collector configuration telemetry section with clearer description and examples:

1. Clearer description for how to configure the `telemetry` section;
2. Extra configuration example for how Collector consumes Collector's own metrics so that we can consume them at the backend. This is a frequently asked question I believe.

Re: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28861

### CODE REVIEW

**Comment by cartermp** [comment]
@open-telemetry/collector-approvers can you please take a look?

File: content/en/docs/collector/configuration.md [hunk 1]
```diff
@@ -574,6 +583,39 @@ service:
       address: 0.0.0.0:8888
 ```
 
+Please note that it's possible to scrape the Collector's metrics by using a
+[`Prometheus` receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver)
```
Comment (chalin) [inline content/en/docs/collector/configuration.md:587]:
```suggestion
Note that it's possible to scrape the Collector's metrics by using a
[Prometheus receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver)
```

File: content/en/docs/collector/configuration.md [hunk 2]
```diff
@@ -546,21 +546,30 @@ service:
...
 
 The `metrics` subsection allows configuration of the metrics generated by the
-collector. By default the collector will generate basic metrics about itself and
-expose them for scraping at `localhost:8888/metrics`
-[View the full list of `metrics` options here.](https://github.com/open-telemetry/opentelemetry-collector/blob/7666eb04c30e5cfd750db9969fe507562598f0ae/config/service.go#L99-L111)
+Collector.
+
+By default, the Collector will generate basic metrics about itself and its
+components, and expose them on port `8888` and path `/metrics`, in Prometheus
+format, so that we can scrape them at `http://localhost:8888/metrics`.
```
Comment (chalin) [inline content/en/docs/collector/configuration.md:567]:
```suggestion
By default, the Collector generates basic metrics about itself and its
components in Prometheus format; you can scrape them at <http://localhost:8888/metrics>.
```

File: content/en/docs/collector/configuration.md [hunk 3]
```diff
@@ -546,21 +546,30 @@ service:
...
 
-The following is an example telemetry configuration:
+View the full list of the `metrics` configuration options
+[here](https://github.com/open-telemetry/opentelemetry-collector/blob/7666eb04c30e5cfd750db9969fe507562598f0ae/config/service.go#L99-L111).
```
Comment (chalin) [inline content/en/docs/collector/configuration.md:570]: I know that this was part of the text before, but I'd suggest dropping this link into a repo source file pinned at a specific commit. That is brittle. Besides, there are only two config options, both of which are illustrated in this page.

```suggestion
```

**Review by chalin — COMMENTED** [review]
Hi. Thanks for the updates. See inline comments.

File: content/en/docs/collector/configuration.md (hunk 2 above)
Comment (brightzheng100) [reply content/en/docs/collector/configuration.md:567]: Originally, the doc mentions that the port `8888` is exposed locally but this is NOT true after checking out the code [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/unmarshaler.go#L63). So I'd suggest to make it clearer that the metrics are exposed at port `8888` for all interfaces by default (which I'm still trying to confirm whether it's aligned with the design in issue [#28861](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28861)) but can be configured in the `telemetry` section.

So I'd assume that the code is correct and may enhance this further if it's agreeable with something like this:
```
By default, the Collector generates basic metrics about itself and its
components, and exposes them in Prometheus format at `:8888/metrics`
of all network interfaces, which can be further controlled in `telemetry`'s `metrics`
subsection; you can scrape them at `http://localhost:8888/metrics`.
```

File: content/en/docs/collector/configuration.md (hunk 3 above)
Comment (brightzheng100) [reply content/en/docs/collector/configuration.md:570]: Do you think it's a good idea to exact these options out for both `logs` and `metrics` sections from the code and make them part of the doc with the necessary explanation?

**Comment by theletterf** [comment]
We might want to merge https://github.com/open-telemetry/opentelemetry.io/pull/3443 first.

**Comment by brightzheng100** [comment]
Sure, I'll rebase to accommodate those changes, after the said PR is merged.

> Note: I was trying to prepare another PR as per stated here https://github.com/open-telemetry/opentelemetry.io/issues/3488, but I didn't expect I would send another PR in such a short timeframe so I inappropriately used the `main` branch for the current PR. Anyway, will wait for this PR to be closed first.

**Comment by svrnm** [comment]
> We might want to merge #3443 first.

This PR has been merged, so you can fix those conflicts now @brightzheng100

**Review by theletterf — APPROVED** [review]

File: content/en/docs/collector/configuration.md (hunk 2 above)
Comment (svrnm) [reply content/en/docs/collector/configuration.md]: @jpkrohling @brightzheng100 can this conversation be resolved?

File: content/en/docs/collector/configuration.md (hunk 3 above)
Comment (svrnm) [reply content/en/docs/collector/configuration.md]: @jpkrohling @brightzheng100 can this conversation be resolved?

**Comment by svrnm** [comment]
@open-telemetry/collector-approvers PTAL!

File: content/en/docs/collector/configuration.md [hunk 4]
```diff
@@ -552,38 +552,99 @@ service:
...
+level of `INFO`. You can also add static key-value pairs to all log entries with
+the `initial_fields` to enrich the logging context. As per the `LogsConfig`
+defined in v{{%
+param vers %}} [here](https://github.com/open-telemetry/opentelemetry-collector/blob/v{{%
+param vers %}}/service/telemetry/config.go), the `logs` configuration options are:
+
+- `level`: sets the minimum enabled logging level, default `INFO`.
+- `development`: puts the logger in development mode, default `false`.
+- `encoding`: sets the logger's encoding, default `console`. Example values are
+  "json", "console".
```
Comment (svrnm) [inline content/en/docs/collector/configuration.md:570]:
~~~suggestion
  `json`, `console`.
~~~

**Comment by svrnm** [comment]
🎉 thank you @brightzheng100

**Pull request merged**